libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go

```diff
@@ -769,24 +776,47 @@ func (dhtPeer *DHTPeer) handleNewDelegationConnection(conn net.Conn) {
 			nbrConns.Dec()
 			break
 		}
+		if envel.Sender != addr {
+			err = errors.New("Sender (" + envel.Sender + ") must match registered address")
+			lerror(err).Str("addr", addr).
+				Msg("while routing delegate client envelope")
+			continue
+		}
 
-		// route envelope
-		dhtPeer.goroutines.Add(1)
-		go func() {
-			defer dhtPeer.goroutines.Done()
-			if envel.Sender != addr {
-				err = errors.New("Sender (" + envel.Sender + ") must match registered address")
-				lerror(err).Str("addr", addr).
-					Msg("while routing delegate client envelope")
-			} else {
-				err := dhtPeer.RouteEnvelope(envel)
-				if err != nil {
-					lerror(err).Str("addr", addr).
-						Msg("while routing delegate client envelope")
-					// TODO() send error back
+		// initializing pairwise channel queues and one go routine per pair
+		pair := envel.To + envel.Sender
+		if _, ok := dhtPeer.syncMessages[pair]; !ok {
+			dhtPeer.syncMessages[pair] = make(chan *aea.Envelope, 100)
```

Comment:
The buffer value is somewhat arbitrary. It needs to be sufficiently high to allow for some buildup.

Reply:
100 is small, bump it to atleast a 1000